### PR TITLE
feat(plugin): commit schema/row snapshots for seven providers

### PR DESCRIPTION
**Pull request by enriquea**
Follow-up to #210, which added the contract ratchet. This backfills the write-verify loop for the providers whose builders already run end-to-end from a committed fixture.

**Ledger: 15 incomplete datasets → 12.** clingen, gencc and hgnc now ship a complete artifact set and drop off entirely; dbnsfp, ensembl-gene, gevir and gnomad-metrics lack only a drift fingerprint (which needs a live upstream probe — deliberately out of scope here).

## What's in it

- **7 snapshot round-trip tests** (clingen, gencc, hgnc, dbnsfp, ensembl-gene, gevir, gnomad-metrics), following the committed `clinvar/tests/test_builder.py` template.
- **4 manifest `fixture:` paths corrected.** They declared a plugin-local directory that was never created, while the tests read a fixture under `hvantk/tests/testdata/raw/`. Moving the fixtures into each plugin was considered and rejected — each is shared by 2–3 test modules (dbNSFP's is also used by an unrelated psroc integration test), so a move would either duplicate data or break other consumers.
- **5 missing `tests/conftest.py` shims.** Those plugins' hail-marked tests could never see the shared `hail_session` / `regenerate_snapshots` fixtures: pytest's conftest discovery only walks *ancestor* directories, and `hvantk/skills/` is a sibling of `hvantk/tests/`. This is why the first regeneration run produced snapshots for only 2 of 7 providers.

## On the sample keys

They're extracted from **real builds**, not invented — `collect_sample_rows` raises `KeyError` for any key absent from the table, so a guessed key fails loudly. The extraction doubled as an independent check that every builder runs through `run_builder_for_spec`:

| provider | rows | key |
|---|---|---|
| dbnsfp | 4,999 | `(locus, alleles)` |
| ensembl-gene | 86,402 | `gene_id` |
| gevir | 19,361 | `gene_id` |
| gnomad-metrics | 542 | `gene_id` |
| gencc | 15 | `(hgnc_id, mondo_id, submitter)` |
| clingen | 11 | `(hgnc_id, mondo_id)` |
| hgnc | 5 | `hgnc_id` |

The snapshots also pin a real behavioural detail worth keeping honest: clingen/gencc store CURIE-stripped ids (`hgnc_id="1100"`) while hgnc:lookup keeps the prefix (`"HGNC:1100"`).

## Verification

Run on Slurm (Hail 0.2.137, Java 11):

| Check | Result |
|---|---|
| Contract subset, snapshots **asserting** (no `--regenerate-snapshots`) | **68 passed, 0 failed** |
| Ratchet | 14 passed |
| flake8 CI gate | 0 |

Asserting rather than regenerating is the point — a regeneration pass only proves files were written.

## Not included, and why

- **uniprot-ptm** — its round-trip test *synthesizes* a TSV into `tmp_path` rather than reading a committed fixture, so there is no fixture to snapshot. Creating one is a separate task.
- **Drift fingerprints** for the four remaining — these need a probe run against the live upstream, which the snapshot tests deliberately never touch.

## Unrelated flake seen en route

`ancestry/test_{pca,pipeline}_hail.py::test_variance_explained` failed in one run with `assert sum(var_exp) <= 1.0` on `1.0000000000000002` — components that sum to 1.0 in exact arithmetic landing two ULPs over. It passed on re-run, and this branch touches nothing in ancestry. It is a pre-existing missing-tolerance bug in the assertion, not a regression here, and it is outside the scoped CI job. Flagging rather than fixing to keep this PR's scope clean.

## Summary by CodeRabbit

- **Tests**
  - Added snapshot-based validation for ClinGen, dbNSFP, Ensembl Gene, GenCC, GeVIR, gnomAD Metrics, and HGNC datasets.
  - Tests now verify generated schemas and representative records remain consistent over time.
  - Added support for regenerating expected snapshots when source data or output formats intentionally change.

- **Chores**
  - Corrected test data references across several dataset integrations.
  - Improved test setup so shared fixtures and validation behavior work consistently across plugin test suites.
  - Updated artifact tracking to reflect newly available validation snapshots.

### CODE REVIEW

File: hvantk/skills/hgnc/tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+top-level test conftest and pytest only needs it registered once across the
+session. Re-exporting it would raise ``ValueError: option names already added``
+when both conftests load (e.g., when running the full suite).
+"""
+
+from hvantk.tests.conftest import (  # noqa: F401
+    hail_session,
+    pytest_collection_modifyitems,
+    regenerate_snapshots,
+)
```
[the remark this answers is not in the document]
Comment (enriquea) [reply hvantk/skills/hgnc/tests/conftest.py:19]: Confirmed, and fixed — though not the way suggested. This is a real bug I hit myself while building this branch (a Slurm run died with exactly `ValueError: no option named '--regenerate-snapshots'`), so thanks for catching that it survives into the shipped state.

Rather than add a duplicate-swallowing `pytest_addoption` to each plugin conftest, the option moved to a new repo-root `conftest.py`. The rootdir is the repo root (pytest.ini lives there), so a conftest there is an ancestor of both `hvantk/tests/` and `hvantk/skills/` and is always loaded. That fixes all 21 plugins with one file instead of 18, and avoids swallowing registration errors.

Verified both directions: `pytest hvantk/skills/hgnc/tests --regenerate-snapshots` now accepts the flag standalone, and the full-suite path raises no duplicate-registration error.

File: hvantk/skills/dbnsfp/plugin.yaml [hunk 2]
```diff
@@ -24,7 +24,7 @@ datasets:
     skill: SKILL.md
     tests:
       command: pytest hvantk/skills/dbnsfp/tests
-      fixture: tests/testdata/raw/dbnsfp
+      fixture: ../../tests/testdata/raw/dbnsfp
```
[the remark this answers is not in the document]
Comment (enriquea) [reply hvantk/skills/dbnsfp/plugin.yaml:31]: Correct — fixed for all four (`dbnsfp`, `ensembl_gene`, `gevir`, `gnomad_metrics`), which are exactly the manifests this PR touches. They were the only ones of the four that lacked `-m hail`; 11 other plugins already had it, so this brings them in line.

The consequence was concrete rather than theoretical: this PR *adds* hail-marked round-trip tests to these four, so as written their documented command would have run the registration test and silently skipped the new snapshot check.

File: hvantk/skills/hgnc/tests/test_builder.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+"""Snapshot round-trip test for the HGNC builder.
+
+Builds from the committed fixture and asserts the schema and a sample of rows against
+committed snapshots, so a change in builder behaviour or upstream field layout shows up
+as an explicit diff rather than silently.
+
+Regenerate after an intentional change:
+    pytest hvantk/skills/hgnc/tests/test_builder_snapshot.py --regenerate-snapshots
+"""
```
[the remark this answers is not in the document]
Comment (enriquea) [reply hvantk/skills/hgnc/tests/test_builder.py:9]: Agreed — fixed in all seven test docstrings, which now read `pytest <path> -m hail --regenerate-snapshots`.

Same trap as the manifest commands: `pytest.ini` sets `addopts = -m "not slow and not network and not hail and not integration"`, so the hint as written would have deselected the very test it was telling you to regenerate.
